great_expectations/data_context/data_context/base_data_context.py

```diff
@@ -348,8 +348,9 @@ def __init__(
             )
         self._ge_cloud_mode = ge_cloud_mode
         self._ge_cloud_config = ge_cloud_config
-        self._project_config = project_config
-        self._apply_global_config_overrides()
+        self._project_config = self._apply_global_config_overrides(
+            config=project_config
+        )
```

Comment:
This is the main change proposed in this PR. Instead of 

1. loading `project_config` parameter into the `_project_config` property
2. changing the property in-place through `_apply_global_config_overrides()` method

the PR is proposing 
1. taking the `project_config` parameter, and passing it to the`_apply_global_config_overrides()` method.
2. The `_apply_global_config_overrides()` will make a `deepcopy` and perform the overrides.
3. The resulting config is returned by `_apply_global_config_overrides()` and loaded into the `_project_config` object of the DataContext objec.t